source/postprocess/visualization/temperature_anomaly.cc

```diff
@@ -0,0 +1,182 @@
+          }
+      }
+
+      template <int dim>
+      void
+      TemperatureAnomaly<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Visualization");
+          {
+            prm.enter_subsection("Temperature anomaly");
+            {
+              prm.declare_entry ("Number of depth slices","20",
+                                 Patterns::Integer (1),
+                                 "Number of depth slices used to define "
+                                 "average temperature.");
+              prm.declare_entry ("Use maximal temperature for bottom","true",
+                                 Patterns::Bool(),
+                                 "If true, use the specified boundary temepratures as average temperatures at the surface. "
+                                 "If false, extrapolate the temperature gradient between the first and second cells to the surface. "
+                                 "This option will only work for models with a fixed surface temperature. ");
+              prm.declare_entry ("Use minimal temperature for surface","true",
+                                 Patterns::Bool(),
+                                 "Whether to use the minimal speficied boundary temperature as the bottom boundary temperature. "
+                                 "This option will only work for models with a fixed bottom boundary temperature. ");
+
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      template <int dim>
+      void
+      TemperatureAnomaly<dim>::parse_parameters(ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Visualization");
+          {
+            prm.enter_subsection("Temperature anomaly");
+            {
+              n_slices = prm.get_integer("Number of depth slices");
+              extrapolate_surface = !prm.get_bool("Use minimal temperature for surface");
+              extrapolate_bottom = !prm.get_bool("Use maximal temperature for bottom");
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(TemperatureAnomaly,
+                                                  "temperature anomaly",
+                                                  "A visualization output postprocessor that outputs the temperature minus the depth-average of the temperature."
+                                                  "The average temperature is calculated using the ``depth average'' postprocessor and interpolated linearly "
```

Comment:
this sounds like the 'depth average' postprocessor has to be active in order to use this postprocessor, but that does not seem to be true. Would it be more correct to reword that it is calculated using the lateral averaging function that is also used by the depth average postprocessor?

Reply:
Corrected.